### PR TITLE
feat: add examples configurations for multiple projects

### PR DESCRIPTION
**Pull request by jogelin**
Related to #2 

ng test my-app --> run only tests for my-app
ng test my-lib --> run only tests for my-lib
ng test --> run all tests

### CODE REVIEW

File: example/angular.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
         "test": {
           "builder": "@angular-builders/jest:run",
           "options": {
+            "configPath": "src/jest.config.js"
```
Comment (meltedspark) [inline example/angular.json]: If there is no file, the option can be abandoned

File: example/angular.json [hunk 2]
```diff
@@ -109,7 +110,128 @@
...
+        "extract-i18n": {
+          "builder": "@angular-devkit/build-angular:extract-i18n",
+          "options": {
+            "browserTarget": "my-app:build"
+          }
+        },
+        "test": {
+          "builder": "@angular-builders/jest:run",
+          "options": {
+            "configPath": "projects/my-app/jest.config.js",
```
Comment (meltedspark) [inline example/angular.json]: If there is no file, the option can be abandoned

File: example/angular.json [hunk 3]
```diff
@@ -109,7 +110,128 @@
...
+          "configurations": {
+            "production": {
+              "project": "projects/my-lib/ng-package.prod.json"
+            }
+          }
+        },
+        "test": {
+          "builder": "@angular-builders/jest:run",
+          "options": {
+            "configPath": "projects/my-lib/jest.config.js",
```
Comment (meltedspark) [inline example/angular.json]: If there is no file, the option can be abandoned

File: example/package.json [hunk 4]
```diff
@@ -30,17 +30,11 @@
     "@angular/cli": "~6.1.2",
     "@angular/compiler-cli": "^6.1.0",
     "@angular/language-service": "^6.1.0",
-    "@types/jasmine": "~2.8.6",
```
Comment (meltedspark) [inline example/package.json]: Jasmine lib  & types should be retained for e2e to work. Only karma should be removed.
Please refer to the updated README for the details.

File: example/angular.json (hunk 1 above)
Comment (jogelin) [reply example/angular.json]: Done

File: example/angular.json (hunk 2 above)
Comment (jogelin) [reply example/angular.json]: Done

File: example/angular.json (hunk 3 above)
Comment (jogelin) [reply example/angular.json]: Done

**Comment by meltedspark** [comment]
Ok, I have taken a closer look into this and here what I have to say:
1. The only differences between the projects (libs, apps) in terms of Jest are path to the _tsconfig.spec.json_ and the root directory of test matching pattern (you used `__dirname` to define those). Correct me if I'm wrong.
2. The intention of this builder is to give a user zero configuration setup, meaning he is required install this builder, update _angular.json_ and go ahead with the unit tests _unless he has a custom configuration_.

Given all this, if I was inferring the root directory inside the builder and adjusting the configuration accordingly (per run), you could avoid exposing Jest configuration in each project. Am I correct?  

If this is the case, I could implement this functionality and the example app would become much simpler with much less noise. What do you think?

**Comment by jogelin** [comment]
1) true.
The main `/example/jest.config.js` only specify the main configurations and it is used for:
- to be extended by `projects/[projectName]/jest.config.js` where I specify the corresponding `tsconfig.spec` and the `testMatch` to only scope the sub-project.
- it is used also to run jest tests by my IDE (webstorm). If I don't specify this global config, my IDE will not find a config a to used.

2) I understand you point and I agree that it should be the default comportment. I was not too much complaining about one `jest.config` per project because it's the way that Angular were doing with `karma.conf.js` files. 

So, indeed, I think it's a good idea to to use this configuration as default configuration for projects !

2 remarks:
- I think the example should contain at least one custom configuration to show how we can overwrite the current config.
- I will still need the main `/example/jest.config.js` if I want my IDE to run tests internally...(maybe just explain that in the README)

**Comment by jogelin** [comment]
Yes I could...today I just have to right click on the file and run. If the config is not at the default place, I'll have to specify the path of the config by creating a run configuration.

But I completely agree that's not a reason ! Go with default config and if I want a specific jest config, it is my personal config !

**Comment by jogelin** [comment]
Another comment not at all related to this repo (I'll delete it just after):

Do you know [storybook](https://storybook.js.org/) ? 

In my monorepo, I am using it to test my shared components and create some examples. They started an integration with Angular but I think that for the moment it is not really going in the `Angular way`. They use configurations of the first application in the `angular.json` and this is not logic...

I opened a ticket [[Feature] Angular 6 Builder](https://github.com/storybooks/storybook/issues/3995) and I think it could really help a lot of users.

My goal is to make a builder to start/serve/build storybook like another app in the `angular.json`. I am not really an expert in webpack but for the moment I can re-use what they did. I'll start by doing the job in a separate repo but I really want to integrate this builder in their repository and I think they are open for that.

Do you think you could be interested to participate in this ? If yes, could I contact you in another channel than github ?

**Comment by meltedspark** [comment]
Ok, so starting from version `1.1.0-beta.2` multiple projects are supported by default.  
I think it would be nice to have a more comprehensive readme than just `This project was generated with Angular CLI version 6.1.2.`  What it is for, what is the structure, the IDE thing etc.
If you still would like to have some library/app with default configuration, you can go ahead and add it, but it should be a real use case. And of course it's worth mentioning that in readme as well.  
Once you confirm everything works as expected I will release `1.1.0`. Once we have a nice README for this example (and of course you confirm the example works for you) I will merge this pull request into the master.  
Thanks for your help, great job!

**Comment by jogelin** [comment]
Do you really want to update the README of the example ? 
I agree we should explain how the builder is working with multiple projects but I don't think this is the role of the README of the example. I think the main README should explain that, do you agree ?

**Comment by meltedspark** [comment]
I didn't mean to explain how the builder works in the example README. But
it is worth to mention in the example README what's its purpose and
structure. Meaning that it is meant to demonstrate the multi project
abilities of the builder and that for example default app and my-lib are
using default config (zero configuration setup) and my-app is using custom
jest  config.

Agree regarding updating the build README (will do that in the
corresponding pull request).

**Comment by jogelin** [comment]
ok I understand. Can I propose a different approach with the examples ? I would split your exmaple in 2:

- A Simple App
- A Multiple Apps/Libs

I doesn't make any sense to mix them...having a main app and sub apps....

**Comment by meltedspark** [comment]
Couldn't agree more.

**Comment by jogelin** [comment]
I added simple-app and multiple-apps in /examples

I had to merge your branch to verify. hope it will not make conflict.

I stille have only one issue when I run npm run test, it should run all tests and it succeed but I get

```
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

**Comment by meltedspark** [comment]
I think it has something to do with the way `ng test` runs multiple projects.  
I would guess that the following happens:
1. `ng test` runs tests for projects one by one in single process.
2. At the end of each test run `jest` checks whether the process has ended or not
3. Since next test is in progress, `jest` thinks that the previous hasn't finished yet and yells. 

I'm not quite sure how to deal with that, but since it doesn't affect exit code it can be flagged as a minor issue and this pull request can be merged.

**Pull request merged**